### PR TITLE
fix: Replace Notify Client with custom instance

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

Removes `notifications-node-client` and implements a custom instance that creates a custom axios config.

Testing now shows that the detailed error is included in the logs.

Screenshot 2024-05-03 at 1 47 58 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +86,6 @@
     "next": "14.2.3",
     "next-auth": "5.0.0-beta.16",
     "node-fetch": "^3.3.2",
-    "notifications-node-client": "7.0.6",
```
Comment (timarney) [inline package.json:89]: :)

File: app/api/share/route.ts [hunk 2]
```diff
@@ -19,21 +19,17 @@ export const POST = middleware([sessionExists()], async (req, props) => {
...
-            },
-            subject: "Form shared | Formulaire partagé",
-            formResponse: `
+        return sendEmail(email, {
+          application_file: {
+            file: base64data,
+            filename: `${slugify(filename)}.json`,
+            sending_method: "attach",
+          },
+          subject: "Form shared | Formulaire partagé",
```
Comment (timarney) [inline app/api/share/route.ts:31]: Can't remember does the "real subject" come from the template itself?

Seeing the current email subject as `GC Forms | Formulaires GC`

Comment (bryan-robitaille) [reply app/api/share/route.ts:31]: Not for the Template used throughout the app.  We have a personalization holder setup in the Subject of the Template.

**Review by craigzour — DISMISSED** [review]
LGTM! Nice work :)

**Review by timarney — DISMISSED** [review]

**Review by craigzour — APPROVED** [review]

**Pull request merged**